src/core/unit.c

```diff
@@ -2997,6 +2999,18 @@ int unit_deserialize(Unit *u, FILE *f, FDSet *fds) {
 
                         continue;
 
+                } else if (streq(l, "cgroup-realized-mask")) {
+
+                        if (safe_atoi(v, (int *)&u->cgroup_realized_mask) < 0)
+                                log_unit_debug(u, "Failed to parse cgroup-realized-mask %s, ignoring.", v);
+                        continue;
+
+                } else if (streq(l, "cgroup-enabled-mask")) {
+
+                        if (safe_atoi(v, (int *)&u->cgroup_enabled_mask) < 0)
```

Comment:
casting the mask ptr to an int ptr like this doesn't appear right to me. IIRC C is permitted to pick any type it likes for implementing an enum, and doesn't have to use int for that, in which case this would explode.